File: 3147/3147.c
#include <stdio.h>

int main() {
    int h, e, a, o, w, x;
    scanf("%d %d %d %d %d %d", &h, &e, &a, &o, &w, &x);
    
    if (h + e + a + x >= o + w) {
        printf("Middle-earth is safe.\n");
    } else {
        printf("Sauron has returned.\n");
    }

    return 0;
}

File: 3147/3147.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	input := scanner.Text()

	var g, e int

	l := strings.Split(input, " ")
	for i := 0; i < len(l); i++ {
		item, _ := strconv.Atoi(l[i])
		if i > 2 && i < 5 {
			e += item
		} else {
			g += item
		}
	}

	if g > e {
		fmt.Println("Middle-earth is safe.")
	} else {
		fmt.Println("Sauron has returned.")
	}
}
